AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/mrecs/RecyclerViewMrecAdActivity.java

```diff
@@ -0,0 +1,243 @@
+        {
+            this.inflater = activity.getLayoutInflater();
+            this.data = data;
+        }
+
+        @NonNull
+        @Override
+        public RecyclerView.ViewHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType)
+        {
+            ViewHolderType viewHolderType = ViewHolderType.values()[viewType];
+
+            switch ( viewHolderType )
+            {
+                case AD_VIEW:
+                    return new AdViewHolder( inflater.inflate( R.layout.activity_mrec_ad_view_holder, parent, false ) );
+                case CUSTOM_VIEW:
+                    return new CustomViewHolder( inflater.inflate( R.layout.activity_mrec_custom_view_holder, parent, false ) );
+                default:
+                    return null;
+            }
+        }
+
+        @Override
+        public void onBindViewHolder(@NonNull final RecyclerView.ViewHolder holder, final int position)
+        {
+            if ( holder instanceof AdViewHolder )
+            {
+                // Select an ad view to display
+                MaxAdView adView = adViews.get( ( position / AD_INTERVAL ) % AD_VIEW_COUNT );
+
+                // Configure cell with an ad
+                ( (AdViewHolder) holder ).configureWithAdView( adView );
+            }
+            else if ( holder instanceof CustomViewHolder )
+            {
+                ( (CustomViewHolder) holder ).textView.setText( data.get( position ) );
+            }
+        }
+
+        @Override
+        public void onViewRecycled(@NonNull final RecyclerView.ViewHolder holder)
+        {
+            super.onViewRecycled( holder );
+
+            if ( holder instanceof AdViewHolder )
+            {
+                AdViewHolder adViewHolder = (AdViewHolder) holder;
+                ( (ViewGroup) adViewHolder.itemView ).removeView( adViewHolder.adView );
+                adViewHolder.adView = null;
+            }
+        }
+
+        @Override
+        public void onViewDetachedFromWindow(@NonNull final RecyclerView.ViewHolder holder)
+        {
+            super.onViewDetachedFromWindow( holder );
+
+            if ( holder instanceof AdViewHolder )
+            {
+                ( (AdViewHolder) holder ).stopAutoRefresh();
+            }
+        }
+
+        @Override
+        public int getItemCount()
+        {
+            return data.size();
+        }
+
+        @Override
+        public int getItemViewType(final int position)
+        {
+            return ( position % AD_INTERVAL == 0 ) ? ViewHolderType.AD_VIEW.ordinal() : ViewHolderType.CUSTOM_VIEW.ordinal();
+        }
+
+        public class AdViewHolder
+                extends RecyclerView.ViewHolder
+        {
+            MaxAdView adView;
+
+            public AdViewHolder(View itemView)
+            {
+                super( itemView );
+            }
+
+            public void configureWithAdView(MaxAdView adView)
```

Comment:
nit: Android naming convention doesn't include the argument name. We can just rename it to `configure( MaxAdView )`.